components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -96,32 +111,146 @@ class SitePermissionsFeature(
      * @param grantedPermissions the list of [grantedPermissions] that have been granted.
      */
     private fun onContentPermissionGranted(session: Session, grantedPermissions: List<Permission>) {
-        session.contentPermissionRequest.consume {
-            it.grant()
-            // Update the DB
+        session.contentPermissionRequest.consume { request ->
+            request.grant()
+            ioCoroutineScope.launch {
+                storeSitePermissions(request, grantedPermissions, ALLOWED)
+            }
             true
```

Comment:
So far we ignored the checkbox in the mocks. I guess now we would need the "Don't ask again" checkbox to determine whether we would need to save this?

Reply:
Yes, I agree.
I have some questions about how the checkbox should work. I filled this issue https://github.com/mozilla-mobile/android-components/issues/2287 to discuss about it

Reply:
👍 Let's write them down there and then ping Bram.